### DISCUSSION TITLE
How can I create a table template (with typst-tablex)?

### DISCUSSION THREAD

**Discussion by maxigaz** [typst/typst#3121] [Q&A] [answered]
I'm new to Typst and I'm trying to figure out how to create a template for a table, so that its general layout and headers can be reused across multiple instances.

Here's what I've been able to write by defining my own table called `ktable`:

```typ
#import "@preview/tablex:0.0.7": tablex, hlinex

#let ktable(cells) = tablex(
  auto-lines: false,
  columns: 2,
  header-rows: 1,

  map-cells: cell => {
    // Resize and center cells in the first column but ignore the first row
    if cell.x == 0 and cell.y > 0 {
      cell.content = {
        set text(22pt)
        cell.content
      }
    }
    cell
  },

  hlinex(),
  [Col1], [Col2],
  hlinex(),

  cells,

  hlinex(),
)

#ktable(
  [1],
  [#lorem(20)],

  [2],
  [#lorem(20)],

  [3],
  [#lorem(20)],
)
```

This produces the error `unexpected argument`. When I uncomment the contents for all except the first cell, the error goes away.

```typ
// This works
#ktable(
  [1],
  // [#lorem(20)],

  // [2],
  // [#lorem(20)],

  // [3],
  // [#lorem(20)],
)
```

I assume it's because `ktable` expects a single argument. How can I go around this?

**Comment by 8LWXpg** [comment, marked as answer]
You can use [argument sinks](https://typst.app/docs/reference/foundations/arguments/#argument-sinks) and [spreading](https://typst.app/docs/reference/foundations/arguments/#spreading)

```typst
#let ktable(..cells) = tablex(
  auto-lines: false,
  columns: 2,
  header-rows: 1,

  map-cells: cell => {
    // Resize and center cells in the first column but ignore the first row
    if cell.x == 0 and cell.y > 0 {
      cell.content = {
        set text(22pt)
        cell.content
      }
    }
    cell
  },

  hlinex(),
  [Col1], [Col2],
  hlinex(),

  ..cells,

  hlinex(),
)

#ktable(
  [1],
  [#lorem(20)],

  [2],
  [#lorem(20)],

  [3],
  [#lorem(20)],
)
```

![image](https://github.com/typst/typst/assets/105704427/ce6df2eb-e86b-4e2a-b29f-d79334afabef)
